models/src/node/drain.rs

```diff
@@ -0,0 +1,364 @@
+                            continue;
+                        }
+                        Ok(StatusCode::INTERNAL_SERVER_ERROR) => {
+                            event!(
+                            Level::ERROR,
+                            "Error when evicting Pod '{}': '{}'. Check for misconfigured PodDisruptionBudgets. Retrying in {:.2}s.",
+                            pod.name(),
+                            e,
+                            EVICTION_RETRY_INTERVAL.as_secs_f64()
+                        );
+                            sleep(EVICTION_RETRY_INTERVAL).await;
+                            continue;
+                        }
+                        Ok(StatusCode::NOT_FOUND) => {
+                            return Err(error::EvictionError::NonRetriableEviction {
+                                source: kube::Error::Api(e.clone()),
+                                pod_name: pod.name().to_string(),
+                            });
+                        }
+                        Ok(StatusCode::FORBIDDEN) => {
+                            // An eviction request in a deleting namespace will throw a forbidden error.
+                            // `kubectl drain` will still wait for these pods to be deleted; however, kube-rs does not provide granular enough access to
+                            // API error statuses to determine if we can proceed, so we ignore these.
+                            return Err(error::EvictionError::NonRetriableEviction {
+                                source: kube::Error::Api(e.clone()),
+                                pod_name: pod.name().to_string(),
+                            });
+                        }
+                        Ok(_) => {
+                            event!(
+                                Level::ERROR,
+                                "Error when evicting Pod '{}': '{}'.",
+                                pod.name(),
+                                e
+                            );
+                            return Err(error::EvictionError::RetriableEviction {
+                                source: kube::Error::Api(e.clone()),
+                                pod_name: pod.name().to_string(),
+                            });
+                        }
+                        Err(_) => {
+                            event!(
+                                Level::ERROR,
+                                "Received invalid response code from Kubernetes API: '{}'",
+                                e
+                            );
+                            return Err(error::EvictionError::RetriableEviction {
+                                source: kube::Error::Api(e.clone()),
+                                pod_name: pod.name().to_string(),
+                            });
+                        }
+                    }
+                }
+                Err(e) => {
+                    event!(Level::ERROR, "Eviction failed: '{}'. Retrying...", e);
```

Comment:
This return Err rather than actually retry. Maybe add retry here or fix the log?

Reply:
Nevermind, seems the RetriableEviction will be handled by RetryStrategy for retrying